src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -278,6 +279,38 @@ public void HotReloadConfig()
         return false;
     }
 
+    /// <summary>
+    /// Get the pagination limit from the runtime configuration.
+    /// </summary>
+    /// <param name="first">The pagination input from the user. Example: $first=10</param>
+    /// <returns></returns>
+    /// <exception cref="DataApiBuilderException"></exception>
+    public uint GetPaginationLimit(int? first)
+    {
+        this.TryGetConfig(out RuntimeConfig? runtimeConfig);
+        uint defaultPageSize = runtimeConfig!.DefaultPageSize();
```

Comment:
```
        uint defaultPageSize = runtimeConfig!.DefaultPageSize();
        uint maxPageSize = runtimeConfig.MaxPageSize();
```

This section seems to be throwing a null reference exception in the tests sometimes when the runtimeconfig is null here, which is happening when the `ODataASTVisitorUnitTests` are running. Those unit tests are SqlTestBase type and so the initialization of the test fixture should not allow this, and when you manually step through one test at a time, at least for me locally, it always passes. But when running the entire test suite, it can happen.

I got this to always pass locally by changing the above to

```
        uint defaultPageSize = runtimeConfig is null ? 100 : runtimeConfig!.DefaultPageSize();
        uint maxPageSize = runtimeConfig is null ? 100000 : runtimeConfig.MaxPageSize();
```

though it would be nice to know why the test is being flakey here.

Reply:
Thanks for helping with this. Let me fix this.